src/venue/core.cljs

```diff
@@ -240,18 +240,19 @@
            :or {timeout? true}} (<! service-request-ch)]
       (if-let [service-provider (get-in @state [:services service])]
         (let [c (chan)
-              to (timeout (or timeout-ms 5000))
+              to (when timeout? (timeout (or timeout-ms 5000)))
```

Comment:
The code will be simplified by getting rid of `timeout?` altogether and always having a timeout of some value. blocking calls to a remote service are simply a way of saying "Don't tell me about this bug until MUCH MUCH later than is useful"